### PR TITLE
Add vite support

### PR DESCRIPTION
**Pull request by IanVS**
Closes https://github.com/storybookjs/addon-svelte-csf/issues/64

This adds the vite plugin from https://github.com/storybookjs/storybook/blob/next/code/frameworks/svelte-vite/src/plugins/csf-plugin.ts into this repo, and sets up a `viteFinal` to load it on vite storybook projects.  There are a few things I'd like to call out here:

1. Without setting a `target` for `@babel/preset-env`, it was trying to add regeneratorRuntime into the built code, which failed because we're not adding it as a polyfill.  I'm imagining that this PR, if merged, would be for use with storybook 7.0+, which supports Chrome 100+ and Node 14+, so those are the targets I added.  This means it's probably a breaking change for this package, so I wanted to call that out explicitly.
2. I needed to turn off some typescript eslint rules, because this repo does not use typescript.
3. If/when we merge this, I'll remove the vite plugin from the storybook vite builder, so that they're not duplicated.  
4. Dual-builder support is a little bit of a bummer in terms of dependencies.  I added magic-string as a dependency because it's used in the plugin, and it doesn't make sense to have as a peer dep.  The others, I made optional peer deps.  It's safe to assume that vite will be available when viteFinal is run, and `@sveltejs/vite-plugin-svelte` should be as well in a svelte vite project, but I added a little safety check around that one.

I tested this by creating an example svelte project, modifying the vite builder to not inject the plugin and copying its `dist` over to the example, and building this addon and copying it's `dist` over as well.  It worked correctly, but it might be nice to consider some tests or example project in this addon, for testing, as well.  I didn't undertake that here, but can if desired.  

<details>
  <summary>📦 Published PR as canary version: <code>2.0.11--canary.72.ae738ba.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/addon-svelte-csf@2.0.11--canary.72.ae738ba.0
  # or 
  yarn add @storybook/addon-svelte-csf@2.0.11--canary.72.ae738ba.0
  ```
</details>

### CODE REVIEW

**Review by JReinhold — APPROVED** [review]
This looks good to me, I think this is the correct way forward!

I changed the base branch to `next` which caused a few conflicts though.

**Comment by JReinhold** [comment]
@IanVS do you want to take a look at the merge conflicts, or instead target this back to 6.5? To me it looks like the conflicts are mostly about the upgrade to 7.0.

File: package.json [hunk 1]
```diff
@@ -68,20 +68,30 @@
...
     "svelte": "^3.50.0",
     "svelte-jester": "^2.3.2",
     "svelte-loader": "^3.1.2",
-    "typescript": "^3.9.7"
+    "typescript": "^3.9.7",
+    "vite": "^3.1.4"
   },
   "peerDependencies": {
     "@storybook/svelte": ">=6.4.20",
     "@storybook/theming": ">=6.4.20",
```
Comment (IanVS) [inline package.json:80]: Is this correct?  Shouldn't these be 7.0?

**Comment by IanVS** [comment]
Thanks for the nudge @JReinhold, I think this is ready for re-review.  I didn't have a chance to test it all out again.

File: package.json (hunk 1 above)
Comment (JReinhold) [reply package.json:80]: You're right. Fixing in another PR

**Comment by IanVS** [comment]
@JReinhold @j3rem1e either of you care to give this a last lookover?  Should we try merging and getting things ready on the builder-vite side before the upcoming 7.0 beta release?

**Comment by shilman** [comment]
:rocket: PR was released in [`v3.0.0-next.1`](https://github.com/storybookjs/addon-svelte-csf/releases/tag/v3.0.0-next.1) :rocket:

**Pull request merged**